bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/TabMain.java

```diff
@@ -237,6 +246,48 @@ protected void updateBuildConfigCombo(String selectedConfigName)
 
 	}
 
+	@Override
+	public void setDefaults(ILaunchConfigurationWorkingCopy configuration)
+	{
+		super.setDefaults(configuration);
+
+		IProject selectedProject = getSelectedProject();
+		if (selectedProject != null)
+		{
+			ICProject icProject = CCorePlugin.getDefault().getCoreModel().create(selectedProject);
+			initializeCProject(icProject, configuration);
+		}
+		try
+		{
+			configuration.doSave();
+		}
+		catch (CoreException e)
+		{
+			Logger.log(e);
+		}
+	}
+
+	private IProject getSelectedProject()
+	{
+		List<IProject> projectList = new ArrayList<>(1);
+		Display.getDefault().syncExec(() -> {
+			IProject project = EclipseUtil.getSelectedProjectInExplorer();
+			if (project != null)
+				projectList.add(project);
+
+		});
+		try
+		{
+			ICProject[] projects = CoreModel.getDefault().getCModel().getCProjects();
+			projectList.addAll(Stream.of(projects).map(ICProject::getProject).collect(Collectors.toList()));
+		}
+		catch (CModelException e)
+		{
+			Logger.log(e);
+		}
+		return projectList.get(0);
+	}
+
```

Comment:
Can you extract this to a utility method as we are using the same code in multiple places in the debug configuration wizard and other places.

Reply:
done